src/datomish/query.cljc

```diff
@@ -147,19 +150,30 @@
 
      [[:datoms12.e :foo] [:datoms13.e :bar]]
 
-   @param context A Context.
-   @param elements The input clause.
+   @param context A Context, containing elements.
    @return a sequence of pairs."
-  [context elements]
-  (when-not (every? #(instance? Variable %1) elements)
-    (raise "Unable to :find non-variables."))
-  (map (fn [elem]
-         (let [var (:symbol elem)]
-           [(lookup-variable context var) (var->sql-var var)]))
-       elements))
-
-(defn context->sql-clause [context elements]
-  {:select (elements->sql-projection context elements)
+  [context]
+  (let [elements (:elements context)]
+    (when-not (every? #(instance? Variable %1) elements)
+      (raise "Unable to :find non-variables."))
+    (map (fn [elem]
+           (let [var (:symbol elem)]
+             [(lookup-variable context var) (var->sql-var var)]))
+         elements)))
+
+(defn row-transducer [context projection rf]
```

Comment:
Why do you want the `context` here?  Is this just `(map row-mapper)`?  Will it grow?